packages/lib/src/proving.ts

```diff
@@ -0,0 +1,105 @@
+const snarkjs = require("snarkjs");
+const fs = require("fs");
+import { Signature, ZKP, MembershipProofInputs } from "./types";
+import { getPublicInputsFromSignature, generateMerkleProof } from "./inputGen";
+import { publicKeyFromString } from "./utils";
+
+export const proveMembership = async (
+  sig: Signature,
+  pubKeys: string[],
+  index: number,
+  msgHash: bigint,
+  pathToCircuits: string | undefined = undefined
+): Promise<ZKP> => {
+  if (isNode() && pathToCircuits === undefined) {
+    throw new Error(
+      "Path to circuits must be provided for server side proving!"
+    );
+  }
+
+  console.time("T and U Generation");
```

Comment:
It's worth noting that if multiple proofs are being generated, all of these time and timeEnds break. So maybe we should assign a uuid to each proof so we can see different timers while debugging

Reply:
Added to batch proving